packages/react/src/components/cookieConsentCore/translations.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable no-param-reassign */
+
+/**
+ * Retrieves a translation based on the provided key, language, and parameters.
+ *
+ * @param {object} translations - The translations object.
+ * @param {string} key - The translation key.
+ * @param {string} lang - The language code.
+ * @param {object} directions - The language directions.
+ * @param {string} fallbackLang - The fallback language code in case the translation is missing in the provided language.
+ * @param {object} parameters - The parameters used to replace placeholders in the translation.
+ * @return {object} - The translated object.
+ * @return {string} [object.value] - The translated value.
+ * @return {boolean} [object.fallback] - Whether the translation is a fallback.
+ * @return {string} [object.lang] - The language code of the translation.
+ * @return {string} [object.dir] - The text direction of the translation.
+ * @throws {Error} - If the translation is missing for the provided key and language.
+ */
+// eslint-disable-next-line import/prefer-default-export
+export function getTranslation(translations, key, lang, directions, fallbackLang, parameters) {
+  // Debug mode, return key instead of translation
+  if (lang === 'key') {
+    return {
+      value: key,
+      fallback: true,
+      lang,
+      dir: 'ltr',
+    };
+  }
+
+  // Fallback language is English by default
+  if (!fallbackLang) {
+    fallbackLang = 'en';
+  }
+
+  // Normal strings as template strings, use like:
+  // index({ a: 1, b: 2 }, "${a} is smaller than ${b}")
+  // returns: 1 is smaller than 2
+  // https://stackoverflow.com/a/41540381
+  function index(obj, is, value) {
+    if (typeof is === 'string') {
+      is = is.split('.');
+    }
+    if (is.length === 1 && value !== undefined) {
+      obj[is[0]] = value;
```

Comment:
Hmm...during writing the tests I'm unable to cover these lines at all (45-46). In what circumstance will these line be executed @Arkkimaagi ?